src/frequenz/client/reporting/_client.py

```diff
@@ -98,7 +98,7 @@ def __iter__(self) -> Iterator[MetricSample]:
                 )
 
 
-class ReportingApiClient:
+class ReportingApiClient(BaseApiClient[ReportingStub, grpcaio.Channel]):
     """A client for the Reporting service."""
 
     def __init__(self, service_address: str, key: str | None = None) -> None:
```

Comment:
You should accept a URL here instead too, and just blindly pass the `server_url` to the `super().__init__()`. The end user needs to decide if they want to use SSL or not.